From 00c4fdb0017d9c54c94338a4a3fb49cf00d707ac Mon Sep 17 00:00:00 2001
From: Marc L <marcll@vt.edu>
Date: Sun, 27 Aug 2017 09:54:23 -0700
Subject: [PATCH 65/72] qtgui: raised time sink inputs to 12/24 and added GRC
 check for 5/10

This commit raises the number of inputs to gr::qtgui:time_sink_c and f
to 12 and 24, respectively. However, the GRC XML wrapper for the block
has a hard-coded limit of 5/10; so a check was added to ensure the
input field in GRC is restricted to this amount. The future fix should
be to rewrite the XML (or YAML in 3.8) to operate as a for loop.
---
 gr-qtgui/grc/qtgui_time_sink_x.xml    | 2 ++
 gr-qtgui/lib/TimeDomainDisplayPlot.cc | 8 +++++++-
 gr-qtgui/lib/time_sink_c_impl.cc      | 3 +++
 gr-qtgui/lib/time_sink_f_impl.cc      | 3 +++
 4 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/gr-qtgui/grc/qtgui_time_sink_x.xml b/gr-qtgui/grc/qtgui_time_sink_x.xml
index fbee4ca483..144cba60eb 100644
--- a/gr-qtgui/grc/qtgui_time_sink_x.xml
+++ b/gr-qtgui/grc/qtgui_time_sink_x.xml
@@ -931,6 +931,8 @@ $(gui_hint()($win))</make>
     <hide>#if ((int($nconnections()) >= 10 or ($type() == "complex" and int($nconnections()) >= 5)) and not $type.t == "message") then 'part' else 'all'#</hide>
   </param>
 
+  <check>$nconnections &lt;= #if $type() == "complex" then 5 else 10#</check>
+
   <sink>
     <name>in</name>
     <type>$type.t</type>
diff --git a/gr-qtgui/lib/TimeDomainDisplayPlot.cc b/gr-qtgui/lib/TimeDomainDisplayPlot.cc
index 51a1989a9a..e1e8482369 100644
--- a/gr-qtgui/lib/TimeDomainDisplayPlot.cc
+++ b/gr-qtgui/lib/TimeDomainDisplayPlot.cc
@@ -170,8 +170,14 @@ TimeDomainDisplayPlot::TimeDomainDisplayPlot(int nplots, QWidget* parent)
   colors << QColor(Qt::blue) << QColor(Qt::red) << QColor(Qt::green)
 	 << QColor(Qt::black) << QColor(Qt::cyan) << QColor(Qt::magenta)
 	 << QColor(Qt::yellow) << QColor(Qt::gray) << QColor(Qt::darkRed)
+	 << QColor(Qt::darkGreen) << QColor(Qt::darkBlue) << QColor(Qt::darkGray)
+	 // cycle through all colors again to increase time_sink_f input limit
+	 // from 12 to 24, otherwise you get a segfault
+	 << QColor(Qt::blue) << QColor(Qt::red) << QColor(Qt::green)
+	 << QColor(Qt::black) << QColor(Qt::cyan) << QColor(Qt::magenta)
+	 << QColor(Qt::yellow) << QColor(Qt::gray) << QColor(Qt::darkRed)
 	 << QColor(Qt::darkGreen) << QColor(Qt::darkBlue) << QColor(Qt::darkGray);
-
+	 
   // Setup dataPoints and plot vectors
   // Automatically deleted when parent is deleted
   for(int i = 0; i < d_nplots; i++) {
diff --git a/gr-qtgui/lib/time_sink_c_impl.cc b/gr-qtgui/lib/time_sink_c_impl.cc
index 06d0c78e9c..6fe06b74a0 100644
--- a/gr-qtgui/lib/time_sink_c_impl.cc
+++ b/gr-qtgui/lib/time_sink_c_impl.cc
@@ -55,6 +55,9 @@ namespace gr {
 	d_size(size), d_buffer_size(2*size), d_samp_rate(samp_rate), d_name(name),
 	d_nconnections(2*nconnections), d_parent(parent)
     {
+      if(nconnections > 12)
+        throw std::runtime_error("time_sink_c only supports up to 12 inputs");
+
       // Required now for Qt; argc must be greater than 0 and argv
       // must have at least one valid character. Must be valid through
       // life of the qApplication:
diff --git a/gr-qtgui/lib/time_sink_f_impl.cc b/gr-qtgui/lib/time_sink_f_impl.cc
index dc88a0add0..1298c6c422 100644
--- a/gr-qtgui/lib/time_sink_f_impl.cc
+++ b/gr-qtgui/lib/time_sink_f_impl.cc
@@ -57,6 +57,9 @@ namespace gr {
 	d_size(size), d_buffer_size(2*size), d_samp_rate(samp_rate), d_name(name),
 	d_nconnections(nconnections), d_parent(parent)
     {
+      if(nconnections > 24)
+        throw std::runtime_error("time_sink_f only supports up to 24 inputs");
+
       // Required now for Qt; argc must be greater than 0 and argv
       // must have at least one valid character. Must be valid through
       // life of the qApplication:
-- 
2.11.0

